### PR TITLE
Feature/gift certificate security

### PR DESCRIPTION
**Pull request by MarinaKhrolovich**
gift certificate security

### CODE REVIEW

File: gift-certificates-app/web/src/main/java/com/epam/esm/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Override
+    @Bean
+    public AuthenticationManager authenticationManagerBean() throws Exception {
+        return super.authenticationManagerBean();
+    }
+
+    @Bean
+    public FilterRegistrationBean registration() {
```
Comment (cam1x) [inline gift-certificates-app/web/src/main/java/com/epam/esm/config/SecurityConfig.java]: Let's parametrize FilterRegistrationBean

File: gift-certificates-app/web/src/main/java/com/epam/esm/exception/MessageLocal.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    String MESSAGE_SOMETHING_WRONG = "message.somethingWrong";
+    String MESSAGE_ID_MIN = "message.path.id.min";
+
+    String MESSAGE_RESOURCE_NOT_FOUND = "message.resource.notFound";
+    String MESSAGE_RESOURCE_ALREADY_EXISTS = "message.resource.alreadyExists";
+    String MESSAGE_RESOURCE_HAS_LINKS = "message.resource.hasLinks";
+    String MESSAGE_RESOURCE_NO_LINKS = "message.resource.noLinks";
+
+    String MESSAGE_TOKEN_INVALID = "message.token.invalid";
```
Comment (cam1x) [inline gift-certificates-app/web/src/main/java/com/epam/esm/exception/MessageLocal.java:13]: Looks like MESSAGE_TOKEN_INVALID and MESSAGE_TOKEN_EXPIRED not used and could be deleted

File: gift-certificates-app/web/src/main/java/com/epam/esm/security/JwtAuthenticationResponse.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.epam.esm.security;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class JwtAuthenticationResponse {
+
+        private int id;
```
Comment (cam1x) [inline gift-certificates-app/web/src/main/java/com/epam/esm/security/JwtAuthenticationResponse.java:10]: Could you please check spacing (indent of class fields) for this file, look's like it differs from others

File: gift-certificates-app/dao/src/main/resources/properties/application.properties [hunk 4]
```diff
@@ -5,3 +5,4 @@ spring.jpa.properties.hibernate.current_session_context_class=org.springframewor
 
 
 
+
```
Comment (cam1x) [inline gift-certificates-app/dao/src/main/resources/properties/application.properties]: Let's keep only one line in the end of file

File: gift-certificates-app/dao/src/test/java/com/epam/esm/dao/impl/UserDAOImplTest.java [hunk 5]
```diff
@@ -31,29 +29,16 @@
...
+        userExpected.setUsername(NEW_USER);
+        userExpected.setPassword(NEW_USER);
 
-        userExists = new User();
-        userExists.setLogin(USER_EXISTS);
-    }
-
-    @Test
-    @Transactional
-    void add() {
```
Comment (cam1x) [inline gift-certificates-app/dao/src/test/java/com/epam/esm/dao/impl/UserDAOImplTest.java]: Looks like User add(User user) and Optional<User> get(String username) not covered

File: gift-certificates-app/web/src/main/java/com/epam/esm/config/SecurityConfig.java (hunk 1 above)
Comment (MarinaKhrolovich) [reply gift-certificates-app/web/src/main/java/com/epam/esm/config/SecurityConfig.java]: done

File: gift-certificates-app/web/src/main/java/com/epam/esm/exception/MessageLocal.java (hunk 2 above)
Comment (MarinaKhrolovich) [reply gift-certificates-app/web/src/main/java/com/epam/esm/exception/MessageLocal.java:13]: I use these messages in JwtTokenService->validateJwtToken. And than JwtAuthenticationException is caugth in GlobalExceptionHandle: exception.getMessage().
Is it better to do it another way?

File: gift-certificates-app/web/src/main/java/com/epam/esm/security/JwtAuthenticationResponse.java (hunk 3 above)
Comment (MarinaKhrolovich) [reply gift-certificates-app/web/src/main/java/com/epam/esm/security/JwtAuthenticationResponse.java:10]: done

File: gift-certificates-app/dao/src/main/resources/properties/application.properties (hunk 4 above)
Comment (MarinaKhrolovich) [reply gift-certificates-app/dao/src/main/resources/properties/application.properties]: done

File: gift-certificates-app/dao/src/test/java/com/epam/esm/dao/impl/UserDAOImplTest.java (hunk 5 above)
Comment (MarinaKhrolovich) [reply gift-certificates-app/dao/src/test/java/com/epam/esm/dao/impl/UserDAOImplTest.java]: done

File: gift-certificates-app/web/src/main/java/com/epam/esm/exception/MessageLocal.java (hunk 2 above)
Comment (cam1x) [reply gift-certificates-app/web/src/main/java/com/epam/esm/exception/MessageLocal.java:13]: Oh, sorry, I missed it. Please ignore my comment then

**Review by cam1x — APPROVED** [review]

**Pull request merged**